### ISSUE TITLE
Let -s+a+r output records with at least one NaN

### ISSUE THREAD

**Issue by seisman** [GenericMappingTools/gmt#3946]
**Description of the desired feature**

Example input dat:
```
1 1 1   0
2 2 NaN 0
3 3 3   NaN
```

Both **-s+a** and **-s+r** work 
```
gmt convert input.dat -s+a
1	1	1	0
gmt convert input.dat -s+r
2	2	NaN	0
```

but **-s+a+r** doesn't work:
```
gmt convert input.dat -s+a+r
1	1	1	0
```
my expected output is:
```
gmt convert input.dat -s+a+r
2 2 NaN 0
3 3 3   NaN
```

**Comment by PaulWessel** [comment]
As implemented, **+r** and **+a** are currently mutually exclusive, so that is the key reason why your case dos not work.  However, I was thinking -s2:3+r would work but nothing comes out because both cols must have NaNs.  So **-s** could use some checking. I guess these are the goals (and I added a 4th record to your input.dat which is 4 4 NaN NAN).

**-s** by itself means **-s**2 and will skip records whose _col_ = 2 column is NaN.  This works
**-s**_col_ means skip records whose column _col_ is NaN. This also works
**-s**2:3 or **-s**2,3 should skip records whose columns 2 **and** 3 is NaN. Not working
**-s**2:3+r*should return records whose columns 2 **and** 3 are NaN.  This is working apparently.
**-s**2:3+a should skip record whose columns 2 **or** 3 are NaN. Seems to work.
**-s**2:3+a+r should pass record whose columns 2 **or** 3 are NaN. Fails because **+r+a** together is not handled.

So the principles are

- The columns considered in the test is limited to column 2, unless _cols_ are specified.  However if _cols_ is **not** specified and **+a** is set then **all** columns shall be considered in the test.
- If the cols specified are **all** NaN then we skip or keep, depending on **+r**
- If **+a** is set then **any** NaN found among the selected columns means we skip or keep, depending on **+r**

So we should have

1. By default, individual column-tests are combined via a logical **AND** across all active columns but **+a** changes this to a logical **OR**.
2. Active columns is only col 2 or all (with **+a**) unless cols is specified
3. The **+r** should invert the result of the test.

Agree, @seisman ?

**Comment by seisman** [comment]
Yes, I believe these are exactly what I (as a user) expect.

> -s2:3 or -s2,3 should skip records whose columns 2 and 3 is NaN. Not working

This one outputs the first 3 records. It seems working to me.

**Comment by seisman** [comment]
Closed by #3956.
